pkg/config/kube/conversion.go

```diff
@@ -72,9 +83,14 @@ func ConvertProtocol(name string, proto coreV1.Protocol) protocol.Instance {
 			name = name[:i]
 		}
 		p := protocol.Parse(name)
-		if p != protocol.UDP && p != protocol.Unsupported {
+		if p != protocol.UDP {
 			out = p
 		}
 	}
+
+	// For well known ports, using protocol sniffing is unnecessary
+	if proto, has := wellKnownPorts[port]; has && out == protocol.Unsupported {
+		out = proto
```

Comment:
I think the problem here is that in the listener code, you are doing merges between the TCP and auto listeners.. instead of declaring it as a conflict. You might need to tweak your logic a bit more in the listener code and use this well known port stuff in listener code instead of kube.conversion.. in listener code, if there is a TCP listener on well known port, you cannot add any http filter chain with the inspector as it will break for server-first protocols. Similarly, if there was a HTTP listener on well known port, you cannot add a tcp listener with the inspector as inspector breaks all server-first protocols.

Reply:
I agree to add the well know port check in the listener. I think we should also have well know port check here. If user don't set protocol for port 3306 of a service and the service is a MySQL server. The protocol sniffing will break server-first protocols too.

Reply:
this line should be out = protocol.TCP as we dont want to enable mysql/redis filters without user consent.